x-pack/solutions/security/plugins/security_solution/public/plugin.tsx

```diff
@@ -183,23 +185,32 @@ export class Plugin implements IPlugin<PluginSetup, PluginStart, SetupPlugins, S
       category: 'admin',
     });
 
-    management?.sections.section.kibana.registerApp({
-      id: 'securityAiAssistantManagement',
-      title: ASSISTANT_MANAGEMENT_TITLE,
-      hideFromSidebar: true,
-      order: 1,
-      mount: async (params) => {
-        const { renderApp, services, store } = await mountDependencies();
-        const { ManagementSettings } = await this.lazyAssistantSettingsManagement();
 
-        return renderApp({
-          ...params,
-          services,
-          store,
-          usageCollection,
-          children: <ManagementSettings />,
+    productFeatureKeys$.pipe(withLatestFrom(plugins.licensing.license$)).subscribe(([productFeatureKeys, license]) => {
+
+      const isInProductFeatureKeys = productFeatureKeys?.has(ProductFeatureAssistantKey.assistant);
+      const hasRegistered = management?.sections.section.kibana.getApp('securityAiAssistantManagement')
+
+      if (isInProductFeatureKeys && license?.hasAtLeast('enterprise') && !hasRegistered) {
```

Comment:
This `isInProductFeatureKeys` name is a bit ambiguous, consider renaming the variables a bit so the condition is easier to understand:

```suggestion
      const isAssistantAvailable = productFeatureKeys?.has(ProductFeatureAssistantKey.assistant) && license?.hasAtLeast('enterprise');
      const assistantManagementApp = management?.sections.section.kibana.getApp('securityAiAssistantManagement')

      if (isAssistantAvailable && !assistantManagementApp) {
```

Reply:
By the way, I see management apps have a `disable` function. Would it make sense to always call the `registerApp` as before, and inside this `subscribe` we only disable it with:

```ts
if (!isAssistantAvailable) {
  assistantManagementApp?.disable();
}
```

Not sure what the `disable` call does, but if that works I think it's better.

wdyt?